finagle-redis/src/main/scala/com/twitter/finagle/redis/GeoCommands.scala

```diff
@@ -0,0 +1,163 @@
+package com.twitter.finagle.redis
+
+import java.lang.{Double => JDouble, Long => JLong, Boolean => JBoolean, Integer => JInt}
+
+import com.twitter.finagle.redis.protocol._
+import com.twitter.finagle.redis.util.BufToString
+import com.twitter.io.Buf
+import com.twitter.io.Buf.Utf8
+import com.twitter.util.Future
+
+private[redis] trait GeoCommands {
+  self: BaseClient =>
+
+  /**
+    * Adds the specified geospatial items (latitude, longitude, name) to the specified `key`
+    *
+    * @return Future of the number of elements added to the sorted set
+    */
+  def geoAdd(key: Buf, values: (Double, Double, Buf)*): Future[JLong] =
+    doRequest(GeoAdd(key, values: _*)) {
+      case IntegerReply(n) => Future.value(n)
+    }
+
+  /**
+    * Get valid Geohash strings representing the position of elements
+    *
+    * @return Future of an array where each element is the Geohash corresponding
+    *         to each member, or `None` if member is missing.
+    */
+  def geoHash(key: Buf, members: Buf*): Future[Seq[Option[Buf]]] =
+    doRequest(GeoHash(key, members: _*)) {
+      case EmptyMBulkReply => Future.Nil
+      case MBulkReply(hashes) => Future.value(hashes.map {
+        case EmptyBulkReply => None
+        case BulkReply(hashBuf) => Some(hashBuf)
+      })
+    }
+
+  /**
+    * Get the positions (longitude,latitude) of all the specified members
+    * of the geospatial index represented by the sorted set at `key`
+    *
+    * @return Future of an array where each element is a tuple representing
+    *         longitude and latitude (x,y) of each member, or `None` if member is missing.
+    */
+  def geoPos(key: Buf, members: Buf*): Future[Seq[Option[(JDouble, JDouble)]]] = {
+    doRequest(GeoPos(key, members: _*)) {
+      case EmptyMBulkReply => Future.Nil
+      case MBulkReply(positions) => Future.value(positions.map {
+        case MBulkReply(Seq(longitude, latitude)) => (longitude, latitude) match {
+          case (BulkReply(lon), BulkReply(lat)) =>
```

Comment:
You can inline `Buf.Utf8(lon)` in this pattern matching to get the string out of `BulkReply(lon)`. This way, you won't need to call  `BufToString` later.